packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -20,34 +20,87 @@ import {
 } from '@react-spectrum/utils';
 import {FocusableRef} from '@react-types/shared';
 import {FocusRing} from '@react-aria/focus';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
 import {mergeProps} from '@react-aria/utils';
-import React, {ElementType, ReactElement} from 'react';
+import {ProgressCircle} from '@react-spectrum/progress';
+import React, {ElementType, ReactElement, useEffect, useState} from 'react';
 import {SpectrumButtonProps} from '@react-types/button';
+import {SpectrumProgressCircleProps} from '@react-types/progress';
 import styles from '@adobe/spectrum-css-temp/components/button/vars.css';
 import {Text} from '@react-spectrum/text';
 import {useButton} from '@react-aria/button';
 import {useHover} from '@react-aria/interactions';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
 import {useProviderProps} from '@react-spectrum/provider';
 
+function disablePendingProps(props) {
+  // Don't allow interaction while isPending is true
+  if (props.isPending) {
+    props.onPress = undefined;
+    props.onPressStart = undefined;
+    props.onPressEnd = undefined;
+    props.onPressChange = undefined;
+    props.onPressUp = undefined;
+    props.onFocus = undefined;
+    props.onBlur = undefined;
+    props.onFocusChange = undefined;
```

Comment:
I think if we remove these, then people may have trouble with cleanup outside the component. It's still focusable, so I think these properties should stay.

On another related note. If someone set isPending during pressstart or keydown... should we still finish the rest of the events? I'm not sure the use case for this, but it seems like you could easily get into trouble if you did this.

should keydown/keyup be disabled as well? they aren't really press and Tab or Esc might be useful, or cleanup of modifier keys?

Reply:
please bring this up in standup/testing as well

Reply:
removed blur and focus events based on feedback